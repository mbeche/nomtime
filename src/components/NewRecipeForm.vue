<script>
import mapMutations from "vuex";
export default {
  name: "NewRecipeForm",
  data() {
    return {
      newRecipeTitle: ""
    };
  },
  methods: {
    ...mapMutations(["ADD_RECIPE"]),
    addRecipe: function() {
      // const recipe = this.newRecipeTitle;
      // this.ADD_RECIPE(recipe);
      this.ADD_RECIPE(this.newRecipeTitle);
      this.newRecipeTitle = "";
    }
  }
};
</script>

<template>
  <div>
    <form @submit.prevent="addRecipe">
      <input
        class="recipe-title"
        type="text"
        placeholder="Recipe Name"
        v-model="newRecipeTitle"
      />
    </form>
  </div>
</template>
