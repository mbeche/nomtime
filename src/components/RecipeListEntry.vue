<script>
export default {
  name: "RecipeListEntry",
  props: {
    recipes: {
      type: Array,
      required: true,
      defualt: () => ({})
    }
  }
};
</script>

<template>
  <div>
    <h2>Recipe List</h2>
    <ul>
      <li v-for="(recipe, index) in recipes" v-bind:key="index">
        <div>Recipe: {{ recipe.title }}</div>
        <div>Ready In: {{ recipe.readyInMinutes }} minutes</div>
      </li>
    </ul>
  </div>
</template>
