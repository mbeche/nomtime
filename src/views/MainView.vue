<script>
export default {
  name: "MainView"
};
</script>

<template>
  <div>MainView</div>
</template>
