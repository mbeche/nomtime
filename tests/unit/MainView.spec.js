import { shallowMount } from "@vue/test-utils";
import MainView from "@/views/MainView";
import RecipeListEntry from "@/components/RecipeListEntry";
import NewRecipeForm from "@/components/NewRecipeForm";

describe("MainView", () => {
  it("renders the component", () => {
    // arrange
    const wrapper = shallowMount(MainView);

    // assert
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders main child components", () => {
    // arrange
    const wrapper = shallowMount(MainView);
    const recipeListEntry = wrapper.find(RecipeListEntry);
    const newRecipe = wrapper.find(NewRecipeForm);

    // assert
    expect(recipeListEntry.exists()).toBe(true);
    expect(newRecipe.exists()).toBe(true);
  });
});
