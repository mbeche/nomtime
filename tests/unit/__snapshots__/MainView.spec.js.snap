// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainView renders the component 1`] = `<div>MainView</div>`;
